src/BooksDisplayer/BooksDisplayer.tsx

```diff
@@ -1,189 +1,236 @@
+      });
+      return outputFilteredBooks;
+    }
+    setFilteredBooks(filterByCategories());
+    setExpandedBook(null);
+  }, [category, records, favorites]);
+
+  useEffect(() => {
+    function filterByName() {
       const nameToLowerCase = name.toLowerCase();
-      const filteredBooks = records.filter(book =>{
-         const title = book.title.toLowerCase();
-         return title.includes(nameToLowerCase);
+      const outputFilteredBooks = records.filter((book: { title: string }) => {
+        const title = book.title.toLowerCase();
+        return title.includes(nameToLowerCase);
       });
 
-      return filteredBooks;
-   }
+      return outputFilteredBooks;
+    }
+    setFilteredBooks(filterByName());
+    setExpandedBook(null);
+  }, [records, name]);
 
-   function handleInputChange(event){
-      setName(event.target.value);
-   }
+  function handleInputChange(event: {
+    target: { value: React.SetStateAction<string> };
+  }) {
+    setName(event.target.value);
+  }
 
+  //Displaying author/authors
+  function displayAuthors(authors: Array<{ name: string }>): string {
+    let output = ``;
+    if (authors.length > 0) {
+      const firstAuthor = authors[0]?.name ?? "";
+      const fullNameOfFirstAuthor = firstAuthor.split(", ");
+      const lastNameOfFirstAuthor = fullNameOfFirstAuthor[0];
+      const firstNameOfFirstAuthor = fullNameOfFirstAuthor[1];
 
-   //Displaying author/authors
-   function displayAuthors(authors){
-      let output = ``;
-      
       if (authors.length > 1) {
-          const firstAuthor = authors[0].name;
-          let [lastNameOfFirstAuthor, firstNameOfFirstAuthor] = firstAuthor.split(", ")
-          
-          const secondAuthor = authors[1].name;
-          let [lastNameOfSecondAuthor, firstNameOfSecondAuthor] = secondAuthor.split(", ");
-          
-          if(firstNameOfFirstAuthor === undefined || firstNameOfSecondAuthor === undefined){
-              firstNameOfFirstAuthor = "";
-              firstNameOfSecondAuthor = ""; 
-          }
-          
-          output = `Authors: ${firstNameOfFirstAuthor} ${lastNameOfFirstAuthor}, ${firstNameOfSecondAuthor} ${lastNameOfSecondAuthor}`;
+        const secondAuthor = authors[1]?.name ?? "";
+        const fullNameOfSecondAuthor = secondAuthor.split(", ");
+        const lastNameOfSecondAuthor = fullNameOfSecondAuthor[0];
+        const firstNameOfSecondAuthor = fullNameOfSecondAuthor[1];
+
+        output = `Authors: ${firstNameOfFirstAuthor} ${lastNameOfFirstAuthor}, ${firstNameOfSecondAuthor} ${lastNameOfSecondAuthor}`;
       } else {
-          const author = authors[0].name;
-          let [lastName, firstName] = author.split(", ");
-          output = `Author: ${firstName} ${lastName}`;
+        output = `Author: ${firstNameOfFirstAuthor} ${lastNameOfFirstAuthor}`;
       }
-      return output;
-   }
+    } else {
+      output = "Unknown authors";
+    }
 
+    return output;
```

Comment:
jak masz opcje to stosuj early return, ładniejszy kodzik
```suggestion
    if (authors.length > 0) {
      const firstAuthor = authors[0]?.name ?? "";
      const fullNameOfFirstAuthor = firstAuthor.split(", ");
      const lastNameOfFirstAuthor = fullNameOfFirstAuthor[0];
      const firstNameOfFirstAuthor = fullNameOfFirstAuthor[1];

      if (authors.length > 1) {
        const secondAuthor = authors[1]?.name ?? "";
        const fullNameOfSecondAuthor = secondAuthor.split(", ");
        const lastNameOfSecondAuthor = fullNameOfSecondAuthor[0];
        const firstNameOfSecondAuthor = fullNameOfSecondAuthor[1];

        return `Authors: ${firstNameOfFirstAuthor} ${lastNameOfFirstAuthor}, ${firstNameOfSecondAuthor} ${lastNameOfSecondAuthor}`;
      } else {
        return `Author: ${firstNameOfFirstAuthor} ${lastNameOfFirstAuthor}`;
      }
    } else {
      return "Unknown authors";
    }
```